src/pages/library/RequestDatasetAccessModal.js

```diff
@@ -60,7 +61,7 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
           ]),
           td([
             Utils.switchCase(access,
-              ['Controlled', () => h(RequestDatasetAccessButton, { datasetName: title })],
+              ['Controlled', () => h(RequestDatasetAccessButton, datasets[0] )],
```

Comment:
```suggestion
              ['Controlled', () => h(RequestDatasetAccessButton, { title, id } )],
```
This way it will change with which button the user clicks, so you can make requests for all of the datasets